Packages/Core/Runtime/Events/AtomEvent.cs

```diff
@@ -104,63 +108,68 @@ public override void UnregisterAll()
         }
 
         /// <summary>
-        /// Register a Listener that in turn trigger all its associated handlers when the Event triggers.
+        /// Register a Listener that in turn trigger all its associated handlers when the Event is invoked.
         /// </summary>
         /// <param name="listener">The Listener to register.</param>
         public void RegisterListener(IAtomListener<T> listener, bool replayEventsBuffer = true)
         {
             _onEvent += listener.OnEventRaised;
             if (replayEventsBuffer)
-            {
                 ReplayBufferToSubscriber(listener.OnEventRaised);
-            }
         }
 
         /// <summary>
-        /// Unregister a listener that was registered using the `RegisterListener` method.
+        /// Unregister a listener that was registered using the <see cref="RegisterListener"/> method.
         /// </summary>
         /// <param name="listener">The Listener to unregister.</param>
         public void UnregisterListener(IAtomListener<T> listener)
         {
             _onEvent -= listener.OnEventRaised;
         }
 
-        #region Observable
         /// <summary>
-        /// Turn the Event into an `IObservable&lt;T&gt;`. Makes Events compatible with for example UniRx.
+        /// Turn the Event into an <see cref="IObservable{T}"/>. Makes Events compatible with for example UniRx.
+        /// </summary>
+        /// <returns>The Event as an <see cref="IObservable{T}"/>.</returns>
+        public IObservable<T> Observe() => new ObservableEvent<T>(Register, Unregister);
+
+        /// <summary>
+        /// Adds <paramref name="item"/> to the replay buffer.
         /// </summary>
-        /// <returns>The Event as an `IObservable&lt;T&gt;`.</returns>
-        public IObservable<T> Observe()
+        /// <param name="item">The item to add.</param>
+        protected void AddToReplayBuffer(T item)
         {
-            return new ObservableEvent<T>(Register, Unregister);
+            if (_replayBufferSize <= 0) return;
+            TrimReplayBuffer();
+            _replayBuffer.Enqueue(item);
         }
-        #endregion // Observable
 
-        protected void AddToReplayBuffer(T item)
+        /// <summary>
+        /// Trim the replay buffer by removing excess items.
+        /// </summary>
+        private void TrimReplayBuffer()
         {
-            if (_replayBufferSize > 0)
-            {
-                while (_replayBuffer.Count >= _replayBufferSize) { _replayBuffer.Dequeue(); }
-                _replayBuffer.Enqueue(item);
-            }
+            while (_replayBuffer.Count >= _replayBufferSize)
+                _replayBuffer.Dequeue();
         }
 
+        /// <summary>
+        /// Replay buffer into a subscriber.
```

Comment:
this is the name of the method ... it does not explain anything.
Wouldn't the more obvious summary be: "Invokes the Action `action` on every element in the replayBuffer" 

btw ... wouldn't the code look cleaner with: `using(var enumerator = ... `
or in other words ... why isn't a foreach loop used here?

this method body could be 3 lines of code

Reply:
I'll change the comment.

I didn't actually look much into what the method was doing TBH. I don't use the replay buffers for anything, so I didn't look at changing it to "using". But I can roll that into a commit.